linkerd/protocol/http/src/test/scala/io/buoyant/linkerd/protocol/http/ZipkinTracePropagatorTest.scala

```diff
@@ -0,0 +1,283 @@
+    //b3 has been removed
+    assert(req.headerMap.get("b3").isEmpty)
+    // check "x-b3-traceid" and "x-b3-spanid" and "x-b3-flags" and "x-b3-parentspanid" have been added to request
+    assert(req.headerMap.keys == Set("x-b3-traceid", "x-b3-spanid", "x-b3-flags", "x-b3-parentspanid"))
+
+    trace match {
+      case Some(tid) =>
+        assert(tid.traceId.toString().equals("a3ce929d0e0e4736"))
+        assert(tid.spanId.toString().equals("00f067aa0ba902b7"))
+        assert(tid.parentId.toString().equals("5b4185666d50f68b"))
+        assert(tid.flags.toLong == 1)
+      case None =>
+        assert(false)  // "traceId does not exist"
+    }
+
+    //test x-b3- headers have been added so we can get the same trace from them
+    val trace2 = ztp.traceId(req)
+    assert(trace == trace2)
+  }
+
+  test("get traceid from a b3 single header - four fields - not sampled child span") {
+    //b3: a3ce929d0e0e4736-00f067aa0ba902b7-d-5b4185666d50f68b
+
+    val ztp = new ZipkinTracePropagator()
+    val req = Request()
+    req.headerMap.add("b3", "a3ce929d0e0e4736-00f067aa0ba902b7-0-5b4185666d50f68b")
+
+    val trace = ztp.traceId(req)
+    //b3 has been removed
+    assert(req.headerMap.get("b3").isEmpty)
+    // check "x-b3-traceid" and "x-b3-spanid" and "x-b3-sampled" and "x-b3-parentspanid" have been added to request
+    assert(req.headerMap.keys == Set("x-b3-traceid", "x-b3-spanid", "x-b3-sampled", "x-b3-parentspanid"))
+
+    trace match {
+      case Some(tid) =>
+        assert(tid.traceId.toString().equals("a3ce929d0e0e4736"))
+        assert(tid.spanId.toString().equals("00f067aa0ba902b7"))
+        assert(tid.parentId.toString().equals("5b4185666d50f68b"))
+        assert(tid.sampled.contains(false))
+      case None =>
+        assert(false)  // "traceId does not exist"
+    }
+
+    // expect to get the right sampled value which is 0
+    // this also checks "x-b3-sampled" has been added to request
+    val sampler = ZipkinTrace.getSampler(req.headerMap)
+    assert(sampler.contains(0.0f))
+
+    //test x-b3- headers have been added so we can get the same trace from them
+    val trace2 = ztp.traceId(req)
+    assert(trace == trace2)
+  }
+
+  test("get traceid from a b3 single header - 128bit trace, two fields") {
+    //b3: 80f198ee56343ba864fe8b2a57d3eff7-05e3ac9a4f6e3b90
+    val ztp = new ZipkinTracePropagator()
+    val req = Request()
+    req.headerMap.add("b3", "80f198ee56343ba864fe8b2a57d3eff7-05e3ac9a4f6e3b90")
+
+    val trace = ztp.traceId(req)
+    //b3 has been removed
+    assert(req.headerMap.get("b3").isEmpty)
+    // check "x-b3-traceid" and "x-b3-spanid" have been added to request
+    assert(req.headerMap.keys == Set("x-b3-traceid", "x-b3-spanid"))
+
+    trace match {
+      case Some(tid) =>
+        assert(tid.traceId.toString().equals("64fe8b2a57d3eff7"))
+        assert(tid.spanId.toString().equals("05e3ac9a4f6e3b90"))
+        assert(tid.traceIdHigh.toString().equals("Some(80f198ee56343ba8)"))
```

Comment:
again, there is no need for Some(), just check with `contains` or any other suitable combinator

Reply:
Done